lib/Parse/ParseIfConfig.cpp

```diff
@@ -299,10 +299,12 @@ class ValidateIfConfigCondition :
       if (E->getArgs()->empty()) {
         D.diagnose(E->getLoc(), diag::platform_condition_expected_argument);
       } else {
-        D.diagnose(E->getLoc(), diag::platform_condition_expected_one_argument);
+        D.diagnose(E->getArgs()->get(1).getStartLoc(),
+                   diag::platform_condition_expected_one_argument);
```

Comment:
The above check for `getSingleSubExpr` can also return `nullptr` for a labelled single argument, e.g:

```swift
#if swift(x: <5)
#endif
```

I guess we probably ought to have a diagnostic for that?

Reply:
Good catch! Since this diagnostic is presumably pretty rare, let's just use `E->getArgs()->front().getStartLoc()` and tweak the message. Added a test case for it